platform/core.windows/src/org/netbeans/core/windows/actions/ResetWindowsAction.java

```diff
@@ -122,6 +122,9 @@ public void run() {
                 } catch( IOException ioE ) {
                     ErrorManager.getDefault().notify( ErrorManager.INFORMATIONAL, ioE );
                 }
+
+                // Reset the "Show Editor Toolbar" setting.
+                EditorOnlyDisplayer.setShowEditorToolbar(true);
```

Comment:
IMO: UI reset should always try to remove config files, so that NB can write fresh ones with new defaults. For multiple reasons: 
 - there are going to be more flags similar to this one here, e.g the editor breadcrumb bar
 - the defaults might change, someone would have to keep this reset method synced with the default config
 - some window visibility toggles seem to be stored somewhere else too https://github.com/apache/netbeans/pull/7496#issuecomment-2189993544 (debugger, eval, breakpont windows don't reset atm)
 - config files could be corrupted, reset might help

The problem however is that this particular setting is stored in preferences, and removing those would require to restart NB.
```java
      // reset editor UI preferences
      if (reset) {
          FileObject conf = FileUtil.getConfigFile("Editors/Preferences/org-netbeans-modules-editor-settings-CustomPreferences.xml");
          if (conf != null) {
              try {
                  conf.delete();
              } catch (IOException ex) {
                  ErrorManager.getDefault().notify(ErrorManager.INFORMATIONAL, ex);
              }
          }
      }
```
(this would restore breadcrumb bar settings etc too after restart)

So I am wondering if we should make `Windows -> Reset Windows` an action which restarts NB so that we can clear more conf files? cc @neilcsmith-net @matthiasblaesing @jtulach

Reply:
i checked if this could be reset to default, but the action always returns false here:
```
toggleEditorToolbar.getValue("preferencesDefault")
```
so this isn't helping.

There would be also `View -> Toolbars -> Reset Toolbars` which is only resetting the main toolbar at the moment.

But since `Reset Windows` is also resetting the main toolbar, I suppose it would be fine if it would enable the editor tool bar too for consistency reasons, without changing any other editor settings (diff sidebar, breadcrumb bar etc)?